File: rec/bin/heat2csv.pl
#!/usr/bin/env perl

use DateTime;
use DateTime::Format::ISO8601;

my $head = 1;
my %MONTH_NAMES = ('Dec' => 12);

while (<>) {

    if (/DataResponse\{(.*)\}/) {

        # --- parse ---
        my $line = $1;
        my $newLine = 1;
        if ($head) {
            print("timestamp");
            foreach my $ii (split(/,\s?/,$line)) {
                my ($name, $rest) = split(/=/, $ii);
                print(", $name");
            }
            $head = 0;
            print("\n");
        }
        foreach my $ii (split(/,\s?/,$line)) {
            my ($name, $rest) = split(/=/, $ii);
            my ($time, $valid, $value) = split(/;/, $rest);
            my ($weekday, $month, $day, $t, $zone, $year) = split(/\s/, $time);
            my ($hour, $minute, $second) = split(/:/, $t);
            my $timestamp = DateTime->new(
                year => $year,
                month => $MONTH_NAMES{$month},
                day => $day,
                hour => $hour,
                minute => $minute,
                second => $second
            );
            if ($newLine) { print("$timestamp") }
            print(", $value");
            $newLine = 0;
        }
        print("\n");
        
        # --- filter ---
        
        # --- sort ---
        
        # --- format ---

    }
}

File: rec/archtools/heat2csv.py

import re
import sys
from datetime import datetime

first_line = True

MONTH_NAMES = {'Dec': 12}

def convert_timestamp(timestamp):
    wday, month, day, hour, minute, sec, tz, year = re.split('\s|:', timestamp)
    dt = datetime(
        int(year),
        MONTH_NAMES[month],
        int(day),
        int(hour),
        int(minute),
        int(sec)
    )
    return dt.isoformat()

for line in sys.stdin:
    match = re.findall('DataResponse\{(.*)\}', line)
    if match:
        signals = match[0].split(',')
        splitted = list([re.split('=|;', sig) for sig in signals])
        if first_line:
            head = ','.join([sig[0] for sig in splitted])
            print('timestamp,{}'.format(head))
            first_line = False
        values = ','.join([sig[3] for sig in splitted])
        timestamp = convert_timestamp(splitted[0][1])
        print('{},{}'.format(timestamp,values))

